src/Data/Vec/Functional.agda

```diff
@@ -108,3 +112,21 @@ zipWith f xs ys i = f (xs i) (ys i)
 
 zip : ∀ {n} → Vector A n → Vector B n → Vector (A × B) n
 zip = zipWith _,_
+
+--------------------------------------------------------------------------------
+--  Combinators
+--------------------------------------------------------------------------------
+
+-- Changes the order of elements in the table according to a permutation (i.e.
+-- an 'Inverse' object on the indices).
+
+permute : ∀ {m n a} {A : Set a} → Fin m ↔ Fin n → Vector A n → Vector A m
```

Comment:
I'd quite like to take this opportunity to correct some sub-optimal design decisions that I think we (well mostly I!) made here.

For starters, the `permute` function should just take a `Fin m → Fin n` function rather than the full inverse proof right? This makes it more generally applicable, and we can then prove properties about it depending on what properties the function satisfies? (also we should try and avoid using the old function hierarchy)